src/js/angular/core/directives/languageselector/language-selector.directive.js

```diff
@@ -31,12 +32,29 @@ function languageSelector($translate, LocalStorageAdapter, LSKeys, $languageServ
             // translated they will be translated in English
             $translate.fallbackLanguage('en');
 
-            $scope.changeLanguage = function (lang) {
-                $scope.selectedLang = lang
+            function setAndPersistLanguage(lang) {
+                $scope.selectedLang = lang;
                 $translate.use(lang.key);
                 LocalStorageAdapter.set(LSKeys.PREFERRED_LANG, lang.key);
                 $languageService.setLanguage($scope.selectedLang.key);
-                $scope.$broadcast('language-changed', {locale: lang.key});
+            }
+
+            $scope.changeLanguage = function (lang) {
+                const eventData = {locale: lang.key, cancel: false};
+                eventEmitterService.emit('language-changed', eventData, (eventData) => {
+                    if (eventData && !eventData.cancel && $languageService.getLanguage() !== lang.key) {
```

Comment:
The check $languageService.getLanguage() !== lang.key is the same in both cases. It is better if this check is made before the 'language-changed' event is emitted; this will prevent unnecessary checks in the yasgui-component directive.

Reply:
OK. I will extract it.